chore: add custom return type lambda for mustache templates

OmarAlJarrah opened the pull request:
## Situation
A new SDK operation was newly proposed which includes PDF file downloads. Due to the fact that our SDK does not support file downloads yet, executing a download operation will result in unexpected behaviour, potentially throwing exceptions on the client-side.

## Task
Add support for file downloads in the SDK. The fastest solution is to return the download stream directly (`InputStream`). It is worth noticing that we need to handle logging for such operations since we do not want to log PDF file bytes.

## Action
* **`customReturnType ` Mustache Lambda**: Added a new Mustache lambda that transforms operations return type of `java.io.File` to `java.io.InputStream`, or returns the original value otherwise. Used the new lambda in place of `returnType`. \
Did not use [type-mapping ](https://openapi-generator.tech/docs/usage/#type-mappings-and-import-mappings) because we do not want this behaviour to be executed globally, but only on the responses level.
* **Loggable Types Resolving**: Added an array of loggable content types, and used it for making the decision of wether to log the body of response/request.

## Result
We now have support for file download operations. The new experience is as follows:
```kotlin
val operation = GetBookingReceiptOperation(params = params)
val stream = client.execute(operation).data // InputStream instance

// example stream consumption...
// read from stream and write to a file.
stream.readAllBytes().let { 
    File("result.pdf").writeBytes(it)
}
```

OmarAlJarrah committed:
chore: add customReturnType lambda for Mustache templates

Introduce customReturnType lambda in MustacheHelpers.kt to handle special return type transformations. Specifically, transforms "java.io.File" to "java.io.InputStream" in generated client methods. Update client.mustache to use the new lambda for return type handling.

OmarAlJarrah committed:
chore: refactor response type handling in Mustache templates

Refactor customReturnType handling logic to streamline return type formatting in client templates. Removed an unnecessary null check and adjusted mustache conditionals to ensure consistent response generation.

OmarAlJarrah committed:
Merge branch 'main' into OmarAlJarrah/add-custom-type-mapping-for-responses

OmarAlJarrah committed:
Merge branch 'main' into OmarAlJarrah/add-custom-type-mapping-for-responses

OmarAlJarrah committed:
Merge branch 'main' into OmarAlJarrah/add-custom-type-mapping-for-responses

OmarAlJarrah committed:
chore: improve request/response body logging

Enhance logging to handle different content types by adding LoggableContentTypes list. Ensure non-loggable content types are appropriately messaged, improving clarity in logs.

OmarAlJarrah committed:
Merge branch 'main' into OmarAlJarrah/add-custom-type-mapping-for-responses

mohnoor94 commented on core/src/main/kotlin/com/expediagroup/sdk/core/plugin/logging/ResponseBodyLogger.kt:
```diff
@@ -26,7 +26,7 @@ import io.ktor.client.plugins.compression.ContentEncoder
 import io.ktor.client.statement.HttpResponsePipeline
 import io.ktor.client.statement.request
-import io.ktor.http.HttpHeaders
+import io.ktor.http.*
```
Let's not use wildcard imports (applicable above too).

OmarAlJarrah committed:
chore: streamline imports in Request/ResponseBodyLogger

Simplify imports by removing unused wildcard imports and adding specific ones. This enhances code readability and maintainability.
